src/stm32f411/clock.cpp

```diff
@@ -0,0 +1,426 @@
+  // 2.1 Move the system_clock to HSI
+  bit_modify(rcc->config).clear(rcc_config::system_clock_switch);
+
+  while (bit_extract<rcc_config::system_clock_switch_status>(rcc->config)) {
+    continue;
+  }
+
+  // 2.2 Disable LSE
+  bit_modify(rcc->backup_domain_control)
+    .clear(rcc_backup_domain_control::low_speed_external_enable);
+
+  while (bit_extract<rcc_backup_domain_control::low_speed_external_ready>(
+    rcc->backup_domain_control)) {
+    continue;
+  }
+
+  // 2.3 Disable HSE
+  bit_modify(rcc->cr).clear(rcc_cr::high_speed_external_enable);
+
+  while (bit_extract<rcc_cr::high_speed_external_ready>(rcc->cr)) {
+    continue;
+  }
+
+  // 2.4 Disable PLL
+  bit_modify(rcc->cr).clear(rcc_cr::main_pll_enable);
+  while (bit_extract<rcc_cr::main_pll_ready>(rcc->cr)) {
+    continue;
+  }
+
+  // =========================================================================
+  // Step 3. Enable External Oscillators
+  // =========================================================================
+  // 3.1 Enable HSE
+  if (p_clock_tree.high_speed_external > 4.0_MHz) {
+    bit_modify(rcc->cr).set(rcc_cr::high_speed_external_enable);
+    while (!bit_extract<rcc_cr::high_speed_external_ready>(rcc->cr)) {
+      continue;
+    }
+  }
+
+  // 3.2 Enable LSE
+  if (p_clock_tree.low_speed_external > 1.0_MHz) {
+    bit_modify(rcc->backup_domain_control)
+      .clear(rcc_backup_domain_control::low_speed_external_enable);
+
+    while (bit_extract<rcc_backup_domain_control::low_speed_external_ready>(
+      rcc->backup_domain_control)) {
+      continue;
+    }
+  }
+
+  // =========================================================================
+  // Step 4. Set up PLL
+  // =========================================================================
+  configure_pll(p_clock_tree);
+
+  // =========================================================================
+  // Step 5. Setup peripheral dividers
+  // =========================================================================
+  bit_modify(rcc->config)
+    .insert(rcc_config::ahb_prescalar, value(p_clock_tree.ahb.divider))
+    .insert(rcc_config::apb1_prescalar, value(p_clock_tree.ahb.apb1.divider))
+    .insert(rcc_config::apb2_prescalar, value(p_clock_tree.ahb.apb2.divider));
+
+  // =========================================================================
+  // Step 6. Setup flash access
+  // =========================================================================
+  bit_modify(flash_config->access_control_reg)
+    .set(flash_acess_control::instruction_cache_en)
+    .set(flash_acess_control::data_cache_en)
+    .set(flash_acess_control::prefetch_cache_en)
+    .insert(flash_acess_control::latency, 2U);
+
+  // =========================================================================
+  // Step 7. Set System Clock and RTC Clock
+  // =========================================================================
+
+  [[maybe_unused]] uint32_t target_clock_source =
```

Comment:
```suggestion
  std::uint32_t target_clock_source =
```

It is used and we should use `std::` for the integer types.